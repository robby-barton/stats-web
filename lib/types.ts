/* istanbul ignore file */

export type YearRanks = {
	weeks: number;
	postseason: boolean;
};

export type AvailRanks = {
	[index: string]: YearRanks;
};

export type Team = {
	team_id: number;
	name: string;
	logo: string;
	logo_dark: string;
};

export type AvailTeams = {
	[index: string]: Team;
};

export type Rank = {
	team: Team;
	conf: string;
	record: string;
	srs_rank: number;
	sos_rank: number;
	final_raw: number;
	final_rank: number;
};

export type TeamRank = {
	team: Team;
	final_rank: number;
	year: number;
	week: string;
	postseason: number;
};

export type TeamGames = {
	team: Team;
	sun: number;
	mon: number;
	tue: number;
	wed: number;
	thu: number;
	fri: number;
	sat: number;
	total: number;
};

export type RankingPath = {
	division: string;
	year: string;
	week: string;
};

export type TeamPath = {
	team: string;
};

export type ChartPoint = {
	week: string;
	rank: number;
	fillLevel: number;
};
