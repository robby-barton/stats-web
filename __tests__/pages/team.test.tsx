import { render, screen } from "@testing-library/react";

import "@testing-library/jest-dom";
import { ChartPoint } from "@lib/types";
import Team from "@pages/team/[team]";

describe("Team page", () => {
	it("renders correctly", async () => {
		const rankList: ChartPoint[] = [
			{
				week: "1",
				rank: 1,
				fillLevel: 25,
			},
		];
		const { baseElement } = render(<Team team="South Carolina" rankList={rankList} years={[2022, 2023]} />);
		await screen.findByTitle(/light mode/i);

		expect(baseElement).toMatchSnapshot();
	});
});
