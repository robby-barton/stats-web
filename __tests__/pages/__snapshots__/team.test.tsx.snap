// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Team page renders correctly 1`] = `
<body
  data-theme="undefined"
>
  <div>
    <div
      class="container"
    >
      <div
        class="header"
      >
        <div
          class="headerTitle"
        >
          <a
            href="/"
          >
            Robby's Ranking
          </a>
        </div>
        <div
          class="headerLink"
        >
          <a
            href="/game-count"
          >
            <span
              class="headingMd"
            >
              Game Count
            </span>
          </a>
        </div>
        <div
          class="headerLink"
        >
          <a
            href="/teams"
          >
            <span
              class="headingMd"
            >
              Teams
            </span>
          </a>
        </div>
        <div
          class="headerLink"
        >
          <a
            href="/about"
          >
            <span
              class="headingMd"
            >
              About
            </span>
          </a>
        </div>
        <div
          class="headerToggle"
        >
          <button
            aria-label="Change to light mode"
            class="css-1f6jnp0"
            title="Change to light mode"
            type="button"
          >
            <span
              class="css-194expn"
            />
            <span>
              üåô
            </span>
            <span>
              ‚òÄÔ∏è
            </span>
          </button>
        </div>
      </div>
      <div
        class="contentWrap"
      >
        <h2>
          South Carolina
        </h2>
      </div>
      <div
        class="footer"
      >
        <a
          href="https://github.com/robby-barton/stats-web"
        >
          &lt;/&gt; - Site source
        </a>
        <br />
        <a
          href="https://github.com/robby-barton/stats-go"
        >
          &lt;/&gt; - Ranking source
        </a>
      </div>
    </div>
  </div>
</body>
`;
